website/management/commands/check_keywords.py

```
@@ -12,32 +13,99 @@ class Command(LoggedBaseCommand):
                 self.stdout.write(self.style.SUCCESS(f"Monitoring {monitor.url}: status {monitor.status}"))
-            except Exception as e:
-                self.stderr.write(self.style.ERROR(f"Error monitoring {monitor.url}: {str(e)}"))
+            except requests.exceptions.Timeout:
+                self.stderr.write(self.style.ERROR(f"Error monitoring {monitor.url}: network request timed out"))
+            except requests.exceptions.ConnectionError:
+                self.stderr.write(self.style.ERROR(f"Error monitoring {monitor.url}: network connection failed"))
+            except requests.exceptions.HTTPError:
+                self.stderr.write(self.style.ERROR(f"Error monitoring {monitor.url}: received non-success HTTP response"))
+            except requests.exceptions.RequestException:
+                self.stderr.write(self.style.ERROR(f"Error monitoring {monitor.url}: network request failed"))
+            except Exception:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)